### ISSUE TITLE
Few issues encountered in tutorial 

### ISSUE THREAD

**Issue by Tianqi-Ma** [smorabit/scWGCNA#21]
Hi, hdWGCNA team:

I was playing around this package recently and found few small issues in tutorial.

1. hdWGCNA Basics
- the Seurat DotPlot section: 
a) the DotPlot function requires "features" argument, or an error will show up:
```
Error in DotPlot(seurat_obj, group.by = "prediction_celltype") : 
  argument "features" is missing, with no default
```
Here is what I add to the DotPlot function:
```
p <- DotPlot(
    seurat_obj,
    features = as.vector(colnames(GetMEs(seurat_obj))),
    group.by = 'prediction_celltype'
```
b) When I make dot plot using the code provided in tutorial, I didn't need to rotate the x/y axis to get the same output (I'm not sure if it's my problem, please double check this).
c) an extra "+" appeared in
```
# flip the x/y axes, rotate the axis labels, and change color scheme:
p <- p +
  coord_flip() +
  RotatedAxis() +
  scale_color_gradient2(high='red', mid='grey95', low='blue') +
```
- the Seurat VlnPlot section:
the "features" argument in VlnPlot function is 'INH-M3', but the output plot tite is 'INH-M12'. 
Is it a mistake?

2. Network Visualization
In "Varying the number of hub genes" section:
```
# plot with ggplot + gganimate
p <- ggplot(df, aes(x=UMAP1, y=UMAP2)) +
  geom_point(color=df$color, size=df$kME*2 ) +
  ggtitle("Supervised weight: {closest_state}") +
  transition_states(
    weight,
    transition_length = 2,
    state_length = 2,
    wrap = TRUE
  ) +
  view_follow() +
  enter_fade() +
  umap_theme()
```
The "weight" should be modified to "n_hubs" and ggtitle should be modified as well.

That's all little mistakes I founded in tutorials. And so far, I only went through these two tutorials. I will post new issues (if there will be) after I go through left tutorials.

I have to say this package is great and the tutorial is remarkably clear to read. Thumbs up for you guys.

I also have few questions/suggestions:

1. It's a really time consuming analysis, especially in the harmony part. In my case, I was planning to do hdWGCNA on all 8 cell types in one scRNA-seq dataset. Is it possible to make it running in parallel in future? BTW, I'm a bit confusing with "group_name" argument in SetDatExpr function. Let's say, if I specify all 8 cell types in "group_name" argument. Will the downstream output be overwrited? And what if I want to save each cell type's plot separately (as I was running analysis on HPC, I want every output plot)? In my case, I can only write a loop and specify only one cell type in each loop and save plots into separated folder to make each cell type's output independent. And if I specify more the 1 cell type in SetDatExpr function, will the functions in Basic Visualization tutorial make plot for every cell type?
2. I used to apply WGCNA before on a tissue development dataset and make a analysis like this (ignore the gene names):
![image](https://user-images.githubusercontent.com/56350087/172824398-06c2c57b-c45d-43b4-997a-99f9597620b7.png)
Basically, I was planning to find the driver genes (those express higher and higher during development), my traits was deveopment time. Perhaps you can make this idea into hdWGCNA as well (or you already done this?)
3. The network looks a bit messy and the labels easily get overlapped. It would be great if the network can be organized. And sometimes it's a bit confusing to save the output plots into png/pdf. My suggestion is to make a argument in every plotting-related function to decided whether generate image file.

Again, I love this package and great works for you guys!

**Comment by smorabit** [comment]
Hi thank you for catching those typos. I would like to draw your attention to this notice at the top of the home page for the package:

> **Note**: hdWGCNA is under active development, so you will likely run into errors if you choose to use hdWGCNA before its first stable release.

The package is still being developed, and I plan to thoroughly proofread the tutorials when a more stable version is released.

Another notice on several of the other tutorials:

> _Tutorial Under Construction_

Which means that I haven't finished writing the tutorial, so use those at your own risk since they are subject to massive changes, and I am not likely to respond to issues regarding these under construction tutorials since they aren't finished.

With regards to your suggestions:

> It's a really time consuming analysis, especially in the harmony part. In my case, I was planning to do hdWGCNA on all 8 cell types in one scRNA-seq dataset. Is it possible to make it running in parallel in future? BTW, I'm a bit confusing with "group_name" argument in SetDatExpr function. Let's say, if I specify all 8 cell types in "group_name" argument. Will the downstream output be overwrited? And what if I want to save each cell type's plot separately (as I was running analysis on HPC, I want every output plot)? In my case, I can only write a loop and specify only one cell type in each loop and save plots into separated folder to make each cell type's output independent. And if I specify more the 1 cell type in SetDatExpr function, will the functions in Basic Visualization tutorial make plot for every cell type?

Parallelization in R is tricky because in general the entire data object needs to be copied for each parallel operation. Meaning in your case that your memory usage would increase by a factor of 8. I don't think this is feasible for most use cases since most people won't have hundreds of gigabytes of memory available. What I do in my own workflow is write a script that runs the hdWGCNA pipeline on each cell population as its own job on my university's HPC.

> I used to apply WGCNA before on a tissue development dataset and make a analysis like this (ignore the gene names):
Basically, I was planning to find the driver genes (those express higher and higher during development), my traits was deveopment time. Perhaps you can make this idea into hdWGCNA as well (or you already done this?)

I don't currently have any plans to make functionality specific for developmental datasets in hdWGCNA. Based on your description it sounds like you want to look into differential gene expression analysis. 

> The network looks a bit messy and the labels easily get overlapped. It would be great if the network can be organized. And sometimes it's a bit confusing to save the output plots into png/pdf. My suggestion is to make a argument in every plotting-related function to decided whether generate image file.

I will take this concern into consideration. hdWGCNA is designed to be quite modular so fortunately you can extract all of the relevant data to make a custom network plot. There is also already an issue (#18) about the overlapping of the labels and it's something I am looking into.
